outer.c

```diff
@@ -207,6 +263,21 @@ static void persist_ns_files(pid_t pid, const char **persist)
    host system, so we reimplement that functionality here. */
 void outer_helper_spawn(struct outer_helper *helper)
 {
+	pid_t rootpid = getpid();
+
+	// If cleanup is needed, fork process to epoll cgroup.events
+	if (helper->cleanfd != -1) {
+		pid_t pid = fork();
+		if (pid == -1) {
+			err(1, "outer_helper: cgroup cleanup fork");
+		}
+
+		if (pid == 0) {
+			cgroup_helper(helper->cleanfd, rootpid);
```

Comment:
cgroup_helper should be ignoring SIGHUP, or calling setsid. This gives the helper a chance to clean up even if the controlling terminal dies.

Reply:
Now calling setsid in the cgroup helper.